### ISSUE TITLE
[bug] accessing variables using environment is not working when using include or preload.

### ISSUE THREAD

**Issue by gust-p** [dynaconf/dynaconf#1029]
**Describe the bug**
env_switcher or environments is not working

**To Reproduce**

1. Having the following folder structure

<details>
<summary> Project structure </summary>

```bash
.
├── .env
├── config
│   ├── .secrets.toml
│   └── settings.toml
│   └── config.yaml
├── db.mysql
├── kpi.py
├── main.py
├── mypy.ini
├── poetry.lock
├── pyproject.toml
└── temporary.py



```
</details>

2. Having the following config files:

<details>
<summary> Config files </summary>

**./.env**
```bash
APP_ENV="production"
```

and

**/config/settings.toml**
```toml
[development]
debug = true
database_url = 'sqlite:///dev.db'

[production]
debug = false
database_url = 'postgres:///prod.db'
```

and 

**/config/config.yaml**
```yaml
location: earth
``` 

</details>

3. Having the following app code:

<details>
<summary> Code </summary>

**./main.py**
```python
from dynaconf import Dynaconf, inspect_settings

envvar_prefix = "APP"
env_switcher = f"{envvar_prefix}_ENV"
default_env = "development"

settings = Dynaconf(
    settings_files=["config/.secrets.toml", "config/settings.toml"],
    includes=["config/config.yaml"],
    enviroments=True,
    load_dotenv=True,
    envvar_prefix=envvar_prefix,
    env_switcher=env_switcher,
)

inspect_settings(settings, print_report=True)

with settings.using_env("production"):
    print(f"Database URL: {settings.DATABASE_URL}")
    print(f"Secret Key: {settings.SECRET_KEY}")

...
```

</details>

4. Executing under the following environment

<details>
<summary> Execution </summary>

```bash
# other commands and details?
# virtualenv activation? yes
python main.py or poetry run python main.py
```

</details>

**Expected behavior**

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04 LSTS
 - Dynaconf Version  3.2.4
 - Frameworks in use: None

**Additional context**
I have tried ```env_switcher=APP_ENV``` and running the app with ```APP_ENV=production python main.py``` or trying ```with settings.using_env()```

**Comment by gust-p** [comment]
Extra context:

When setting ```environments=True``` nothing is accessible, leading to ```AttributeError```. And nothing on the ```config/config.yaml``` file is loaded.

```yaml
header:
  env_filter: None
  key_filter: None
  new_first: 'True'
  history_limit: None
  include_internal: 'False'
current:
  ENVIROMENTS: true
  LOAD_DOTENV: true
  SECRET_KEY: supersecret
  DEVELOPMENT:
    debug: true
    database_url: sqlite:///dev.db
  PRODUCTION:
    debug: false
    database_url: postgres:///prod.db
  ENV: production
history:
- loader: env_global
  identifier: unique
  env: global
  merged: false
  value:
    ENV: production
- loader: toml
  identifier: /home/leno/Repositories/kpi_refactor/config/settings.toml
  env: default
  merged: false
  value:
    DEVELOPMENT:
      debug: true
      database_url: sqlite:///dev.db
    PRODUCTION:
      debug: false
      database_url: postgres:///prod.db
- loader: toml
  identifier: /home/leno/Repositories/kpi_refactor/config/.secrets.toml
  env: default
  merged: false
  value:
    SECRET_KEY: supersecret
- loader: set_method
  identifier: settings_module_method
  env: global
  merged: false
  value:
    SETTINGS_MODULE:
    - config/.secrets.toml
    - config/settings.toml
- loader: set_method
  identifier: envvars_first_load
  env: global
  merged: false
  value:
    ENV_FOR_DYNACONF: production
- loader: set_method
  identifier: init_kwargs
  env: global
  merged: false
  value:
    ENVIROMENTS: true
    LOAD_DOTENV: true
    INCLUDES_FOR_DYNACONF:
    - config/config.yaml
    ENV_SWITCHER_FOR_DYNACONF: APP_ENV
    ENVVAR_PREFIX_FOR_DYNACONF: APP
    SETTINGS_FILE_FOR_DYNACONF:
    - config/.secrets.toml
    - config/settings.toml
```

With ```environments=False``` and ```preload=["config/config.yaml"]```:

```yaml
header:
  env_filter: None
  key_filter: None
  new_first: 'True'
  history_limit: None
  include_internal: 'False'
current:
  LOAD_DOTENV: true
  ENVIROMENTS: false
  LOCATION: earth
  SECRET_KEY: supersecret
  DEVELOPMENT:
    debug: true
    database_url: sqlite:///dev.db
  PRODUCTION:
    debug: false
    database_url: postgres:///prod.db
  ENV: production
history:
- loader: env_global
  identifier: unique
  env: global
  merged: false
  value:
    ENV: production
- loader: toml
  identifier: /home/leno/Repositories/kpi_refactor/config/settings.toml
  env: default
  merged: false
  value:
    DEVELOPMENT:
      debug: true
      database_url: sqlite:///dev.db
    PRODUCTION:
      debug: false
      database_url: postgres:///prod.db
- loader: toml
  identifier: /home/leno/Repositories/kpi_refactor/config/.secrets.toml
  env: default
  merged: false
  value:
    SECRET_KEY: supersecret
- loader: set_method
  identifier: settings_module_method
  env: global
  merged: true
  value:
    SETTINGS_MODULE:
    - config/.secrets.toml
    - config/settings.toml
- loader: yaml
  identifier: /home/leno/Repositories/kpi_refactor/config/config.yaml
  env: default
  merged: false
  value:
    LOCATION: earth
- loader: set_method
  identifier: envvars_first_load
  env: global
  merged: false
  value:
    ENV_FOR_DYNACONF: production
- loader: set_method
  identifier: init_kwargs
  env: global
  merged: false
  value:
    LOAD_DOTENV: true
    ENVIROMENTS: false
    SETTINGS_FILE_FOR_DYNACONF:
    - config/.secrets.toml
    - config/settings.toml
    ENVVAR_PREFIX_FOR_DYNACONF: APP
    PRELOAD_FOR_DYNACONF:
    - config/config.yaml
    MERGE_ENABLED_FOR_DYNACONF: true
    ENV_SWITCHER_FOR_DYNACONF: APP_ENV
```

**Comment by rochacbruno** [comment]
To be addressed on #1157
